### DISCUSSION TITLE
saving the states of open tabs with terminals

### DISCUSSION THREAD

**Discussion by syfulin** [coder/code-server#6938] [Q&A]
Question: does code-server allow saving the states of open tabs with terminals and programs running in them? So that when closing a browser tab with code-server and reopening the tab with terminals and programs running in them, they are restored and their execution is not interrupted.

**Comment by code-asher** [comment]
It should do this, but it is per-browser so switching to another browser/computer will have a new state.

Also, it looks like it kills terminals after a while, so if you need them to stay up long-term you have to use something like tmux.

Also extensions are *not* persisted.  As soon as you close the browser tab the extension host is killed.  This also means debugging state is not restored.
